cmd/root.go

```diff
@@ -256,6 +269,32 @@ func runSignalWrapper(cmd *Command) error {
 
 	shutdownCh := make(chan error)
 
+	if cmd.prometheusNamespace != "" {
+		e, err := prometheus.NewExporter(prometheus.Options{
+			Namespace: cmd.prometheusNamespace,
+		})
+		if err != nil {
+			return err
+		}
+		mux := http.NewServeMux()
+		mux.Handle("/metrics", e)
+		addr := fmt.Sprintf("localhost:%s", cmd.prometheusPort)
+		server := &http.Server{Addr: addr, Handler: mux}
+		go func() {
+			select {
+			case <-ctx.Done():
+				if err := server.Shutdown(context.Background()); err != nil {
+					shutdownCh <- fmt.Errorf("failed to stop prometheus HTTP server: %v", err)
+				}
+			}
+		}()
+		go func() {
+			if err := server.ListenAndServe(); err != nil {
```

Comment:
Should we check for `http.ErrServerClosed` here?

Reply:
Good idea.